ld-elf2flt.c

```diff
@@ -324,6 +324,24 @@ static int do_final_link(void)
 		append_option(&other_options, concat(got_offset, "=", buf, NULL));
 	}
 
+	if (streq(TARGET_CPU, "riscv64")) {
+		/*
+		 * The .got section must come before the .got.plt section
+		 * (gcc/ld bug ?).
+		 */
+		append_sed(&sed, "(.got.plt)", "(.got.tmp)");
+		append_sed(&sed, "(.got.plt)", "(.got)");
+		append_sed(&sed, "(.got.tmp)", "(.got.plt)");
+
+		/*
+		 * The global pointer symbol is defined after the GOT.
+		 */
+		append_sed(&sed, "^RV64_GP", "");
+	} else {
+		/* Get rid of the global pointer definition */
+		append_sed(&sed, "^RV64_GP", NULL);
```

Comment:
iiuc, `__global_pointer$` is not specific to riscv64, but all riscv ABIs utilize it.  so i would name it `RISCV_GP`.